waspc/src/Wasp/AppSpec/App/EmailSender.hs

```diff
@@ -0,0 +1,30 @@
+{-# LANGUAGE DeriveDataTypeable #-}
+
+module Wasp.AppSpec.App.EmailSender
+  ( EmailSender (..),
+    EmailProvider (..),
+    EmailFrom (..),
+  )
+where
+
+import Data.Data (Data)
+
+data EmailSender = EmailSender
+  { provider :: EmailProvider,
+    defaultFrom :: EmailFrom
+  }
+  deriving (Show, Eq, Data)
+
+data EmailProvider = SMTP | SendGrid | Mailgun
+  deriving (Eq, Data)
+
+instance Show EmailProvider where
+  show SMTP = "smtp"
+  show SendGrid = "sendgrid"
+  show Mailgun = "mailgun"
```

Comment:
Why are you doing this?
Since default implementation of Show would already work quite well, and this one only has everything lowercase, I am guessing you had a good reason for doing this -> some other part of code is expecting it to be lowercase.
If that is so, then `Show` is not the best thing to use -> `Show` is meant to be used as a debugging / prettifying thing, not to provide string ids that the rest of the code will rely on.
So if you need string ids, I would rather suggest implementing a function like `toStringId` or something like that.

Reply:
Thank you for the tip! I thought `Show` instance is the right way to do this.

I had some earlier ideas to use this key (lowercase name) to map things in the templates, but I decided not to do it. I can remove this custom impl and just derive `Show` instead.